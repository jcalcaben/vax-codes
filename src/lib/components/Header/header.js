import React from "react";

import {root, brand, scan, nav, navItem } from "./header.module.css";

const Header = (props) => {
  return (
    <header className={root}>
      <span className={brand}>Vax.Codes</span>
      <button className={scan}>Scan QR Code</button>
      <nav className={nav}>
        <span className={navItem}>How it works</span>
        <span className={navItem}>About us</span>
      </nav>
    </header>
  );
};

export default Header;
