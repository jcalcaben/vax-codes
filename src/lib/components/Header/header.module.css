.root {
  composes: max-w-full from global;
  composes: bg-blue-600 from global;
  composes: grid from global;
  composes: grid-cols-3 from global;
  composes: p-2 from global;
  composes: text-white from global;
  composes: items-center from global;
}

.brand {
  composes: text-xl from global;
}

.nav {
  composes: text-blue-300 from global;
  composes: justify-self-end from global;
  composes: justify-items-center from global;
  composes: grid from global;
  composes: grid-cols-2 from global;
}

.navItem {
  /** Hover Styles **/
  composes: hover_text-blue-100 from global;
}

.scan {
  composes: justify-self-center from global;
  composes: border-2 from global;
  composes: p-2 from global;
  composes: rounded-md from global;

  /** Hover Styles **/
  composes: hover_bg-white from global;
  composes: hover_text-blue-600 from global;
}
