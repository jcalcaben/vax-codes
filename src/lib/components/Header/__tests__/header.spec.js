
import React from 'react';
import Header from '../header';
import ShallowRenderer from 'react-test-renderer/shallow';

const renderer = new ShallowRenderer();

describe("Header component", () => {
  test("renders the correct basic shape", () => {
    renderer.render(<Header>Content</Header>);
		const output = renderer.getRenderOutput();

    expect(output).toMatchSnapshot();
  });
});
