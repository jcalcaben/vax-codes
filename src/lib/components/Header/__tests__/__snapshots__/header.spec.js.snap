// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the correct basic shape 1`] = `
<header>
  Header content
</header>
`;
